airbyte/_processors/sql/snowflake.py

```
@@ -37,13 +38,86 @@ class SnowflakeConfig(SqlConfig):
+        auth_methods = {
+            "password": self.password is not None,
+            "private_key": self.private_key is not None,
+            "private_key_path": self.private_key_path is not None,
+        }
+        has_passphrase = self.private_key_passphrase is not None
+        primary_auth_count = sum(auth_methods.values())
+
+        if primary_auth_count == 0:
+            if has_passphrase:
+                raise ValueError(
+                    "You have to provide a primary authentication method if you want to use a private key passphrase."
```

Ruff (0.11.9)

Line too long (118 > 100)

(E501)